go/host/p2p/p2p.go

```diff
@@ -370,6 +354,15 @@ func (p *Service) handle(conn net.Conn) {
 		}
 		// this is an incoming request, p2p service is responsible for finding the response and returning it
 		go p.handleBatchRequest(msg.Contents)
+	case msgTypeRegisterForBroadcasts:
```

Comment:
is there a mechanism to clean up the peers, in case they drop off the network?
Also, what about restarts and persistence?

Reply:
No persistence, I've tried to keep it quite tolerant to validators coming and going with these rules:
- if broadcast to a validator fails 3 times then the validator is removed from the pool (this cleans up if they shut down)
- if validator doesn't receive a broadcast in a 2min period it requests to the sequencer to add it back to the pool (this would account for the sequencer getting restarted, the validators would re-register)

Reply:
Sounds good. But this mechanism should be highlighted or commented on in the code. 
This behaviour was not obvious to me at all.